api/lib/block-listeners/polkadot-node-subscription.js

```diff
@@ -16,15 +19,65 @@ class PolkadotNodeSubscription extends BaseNodeSubscription {
   // the class gets stored in the store to be used by all instances
   async initPolkadotRPC(network, store) {
     const api = new ApiPromise({
-      provider: new WsProvider(
-        network.rpc_url || network.public_rpc_url
-      )
+      provider: new WsProvider(network.rpc_url || network.public_rpc_url)
     })
     store.polkadotRPC = api
     store.polkadotRPCOpened = Date.now()
     await api.isReady
     console.log(network.id + ' API initialized')
   }
+
+  async getValidators(block, dataSource) {
+    const { sessionIndex, era: currentEra } = this.store.data
+    if (sessionIndex < block.sessionIndex || !sessionIndex) {
+      this.store.update({
+        data: {
+          sessionIndex: block.sessionIndex
+        }
+      })
+      console.log(
+        `\x1b[36mCurrent session index is ${block.sessionIndex}, fetching validators!\x1b[0m`
+      )
+      const [validators, era] = await Promise.all([
+        dataSource.getAllValidators(),
+        this.store.polkadotRPC.query.staking.activeEra().then(async (era) => {
```

Comment:
those should probably become getters in polkadot source

Reply:
Done!